src/plugins/EnergyRAPL.h

```diff
@@ -0,0 +1,182 @@
+#ifndef SRC_PLUGIN_ENERGYRAPL_H_
+#define SRC_PLUGIN_ENERGYRAPL_H_
+
+#include <chrono>
+#include <string>
+
+#include "plugins/PluginBase.h"
+
+#ifdef ENABLE_MPI
+#include "mpi.h"
+#endif
+
+/**
+ * @brief This plugin computes the energy consumption of the simulation using RAPL counters via the sysfs interface.
+ * @author Ruben Horn
+ *
+ * The output of this plugin is either written to the info logger or to a tab separated file (.tsv) with the columns
+ * milliseconds, simstep and joules.
+ *
+ * \b NOTE:
+ *  - You must ensure that the files under /sys/class/powercap/intel-rapl/ are readable
+ *  - The total energy is measured over the package and DRAM domains, if present.
+ */
+class EnergyRAPL : public PluginBase {
+private:
+	/**
+	 * @brief Class to measure the energy consumption using the sysfs interface for the corresponding RAPL counter
+	 * domain
+	 */
+	class RAPLCounter {
+	private:
+		/**
+		 * @brief Path to the file containing the current value for this domain
+		 */
+		std::string _microJoulesPath;
+		/**
+		 * @brief Last value measured
+		 */
+		long long _lastMicroJoules;
+		/**
+		 * @brief Maximum counter value (used for overflow detection)
+		 */
+		long long _rangeMicroJoules;
+		/**
+		 * @brief Aggregated value since initialization or reset
+		 */
+		long long _microJoules;
+
+	public:
+		/**
+		 * @brief Creates a new instance of a counter for the energy consumption of a specific domain
+		 *
+		 * @param domainBasePath The sysfs base path of the corresponding RAPL domain (must contain the files: name,
+		 * max_energy_range_uj, energy_uj)
+		 */
+		RAPLCounter(const std::string domainBasePath);
+
+		/**
+		 * @brief Reset the counter for this domain to 0
+		 */
+		void reset();
+
+		/**
+		 * @brief Takes a new measurement for this domain
+		 * @return The aggregated value since initialization or last reset
+		 */
+		double update();
+	};
+
+	/**
+	 * @brief Base path for the RAPL sysfs interface
+	 */
+	const char* const _basePathRAPL = "/sys/class/powercap/intel-rapl/";
+
+	/**
+	 * @brief Counter objects for all RAPL domains used to compute the energy consumption
+	 */
+	std::vector<RAPLCounter> _counters;
+
+	/**
+	 * @brief Number of simulation steps between outputs (only output at the end if 0)
+	 */
+	int _writeFrequency = 0;
+	/**
+	 * @brief Filename without extension for outputs (output to stdout if empty string)
+	 */
+	std::string _outputprefix;
+	/**
+	 * @brief Aggregated energy consumption over all \ref _counters
+	 */
+	double _joules;
+	/**
+	 * @brief Simulation step (updated by plugin hook)
+	 */
+	unsigned long _simstep;
+	/**
+	 * @brief Start time of the simulation (used to compute timestamp in milliseconds)
+	 */
+	std::chrono::_V2::steady_clock::time_point _simstart;
```

Comment:
```suggestion
	std::chrono::steady_clock::time_point _simstart;
```
AFAIK the `_V2`is for internal use only.

From `chrono.h`:
>    // To support the (forward) evolution of the library's defined
>    // clocks, wrap inside inline namespace so that the current
>    // defintions of system_clock, steady_clock, and
>    // high_resolution_clock types are uniquely mangled. This way, new
>    // code can use the latests clocks, while the library can contain
>    // compatibility definitions for previous versions.  At some
>    // point, when these clocks settle down, the inlined namespaces
>    // can be removed.  XXX GLIBCXX_ABI Deprecated
>    inline namespace _V2 {